lib/Sema/CSSimplify.cpp

```diff
@@ -960,9 +960,32 @@ ConstraintSystem::TypeMatchResult constraints::matchCallArguments(
 
   for (unsigned paramIdx = 0, numParams = parameterBindings.size();
        paramIdx != numParams; ++paramIdx){
-    // Skip unfulfilled parameters. There's nothing to do for them.
-    if (parameterBindings[paramIdx].empty())
+    // If a parameter is unfulfilled, validate the default argument if it is a
+    // magic literal expression.
+    if (parameterBindings[paramIdx].empty()) {
+      if (!isa<FuncDecl>(callee))
+        continue;
+
+      auto allParams = dyn_cast<FuncDecl>(callee)->getParameters()->getArray();
+      auto param = allParams[paramIdx];
+
+      assert(param->isDefaultArgument());
+
+      auto defaultValueExpr = param->getDefaultValue();
+
+      if (!isa<MagicIdentifierLiteralExpr>(defaultValueExpr))
+        continue;
+
+      cs.generateConstraints(defaultValueExpr, param->getDeclContext());
```

Comment:
One more final suggestion here - if `defaultValueExpr` already has a type it means that it has been validated so we could just use that type here instead of creating constraints.